### PR TITLE
Added `adjustUrl` function

### PR DESCRIPTION
**Pull request by zdila**
GO-840

## Objective
Added `adjustUrl` function replacing less universal `adjustUrlQuery` function.

## Description
`adjustUrl` receives complete URL which also enables to read and/or update search query.

### CODE REVIEW

**Review by petr-hajek — CHANGES_REQUESTED** [review]

**Comment by zdila** [comment]
example added in 9f3bfe34bf6a706e68dd11059ff6c84237c1515a

File: src/types.ts [hunk 1]
```diff
@@ -309,9 +309,18 @@ export type ControlOptions = {
...
    * Default value is empty function.
+   *
+   * @deprecated use `adjustUrl`
    */
   adjustUrlQuery?: (sp: URLSearchParams) => void;
 
+  /**
+   * Function to adjust geocoding URL before the fetch.
+   *
+   * Default value is empty function.
```
[the remark this answers is not in the document]
Comment (zdila) [reply src/types.ts]: See 9f3bfe34bf6a706e68dd11059ff6c84237c1515a

[the remark this answers is not in the document]
Comment (petr-hajek) [reply src/types.ts]: Thanks, this will probably be better to add to the docs than here

**Review by petr-hajek — APPROVED** [review]
After discussion, we will go ahead with this solution for now, but we will try to figure out some new user-friendly default for reverse geocoding (possibly something to reflect user's zoom level). We'll make a docs to explain to users how to adjust reverse geocoding using adjustUrl function

**Pull request merged**